Cargo.toml

```diff
@@ -209,65 +209,87 @@ diplomat-runtime = { git = "https://github.com/rust-diplomat/diplomat.git", rev
+# Keep in sync with allowlist.rs:
+# BASIC_RUNTIME_DEPS
+# BASIC_BUILD_DEPS
+# EXTRA_SERDE_DEPS
+# EXTRA_EXPERIMENTAL_DEPS
+# EXTRA_LSTM_DEPS
+# EXTRA_RYU_DEPS
+# EXTRA_CAPI_DEPS
+# EXTRA_BLOB_DEPS
+# EXTRA_FS_DEPS
 core_maths = { version = "0.1.0", default-features = false }
 displaydoc = { version = "0.2.3", default-features = false }
 either = { version = "1.9.0", default-features = false }
-elsa = { version = "1.10.0", default-features = false }
-erased-serde = { version = "0.3.11", default-features = false }
-eyre = { version = "0.6.0", default-features = false }
-indexmap = { version = "1.0.0", default-features = false }
-itertools = { version = "0.10.3", default-features = false}
 libc_alloc = { version = "1.0.6", default-features = false }
 log = { version = "0.4.17", default-features = false }
 memchr = { version = "2.6.0", default-features = false }
-ndarray = { version = "0.15.0", default-features = false }
-nom = { version = "7.0.0", default-features = false }
 num-bigint = { version = "0.4.3", default-features = false }
 num-rational = { version = "0.4.0", default-features = false }
 num-traits = { version = "0.2.0", default-features = false }
-once_cell = {version = "1.15.0", default-features = false}
 postcard = { version = "1.0.1", default-features = false }
-proc-macro2 = { version = "1.0.61", default-features = false }
-quote = { version = "1.0.28", default-features = false }
-rayon = { version = "1.3.0", default-features = false }
 regex-automata = { version = "0.2.0", default-features = false }
 ryu = { version = "1.0.5", default-features = false }
 serde = { version = "1.0.110", default-features = false }
-serde_json = { version = "1.0.45", default-features = false, features = ["alloc"] } # requires either alloc or std to build
-serde-aux = { version = "4.0.0", default-features = false }
 serde-json-core = { version = "0.4.0", default-features = false }
-simple_logger = { version = "4.0.0", default-features = false }
 smallvec = { version = "1.10.0", default-features = false }
 stable_deref_trait = { version = "1.2.0", default-features = false }
-syn = { version = "2.0.21", default-features = false }
-synstructure = { version = "0.13.0", default-features = false }
-toml = { version = "0.5.8", default-features = false }
-twox-hash = { version = "1.4.2", default-features = false }
 unicode-bidi = { version = "0.3.11", default-features = false }
-ureq = { version = "2.3.0", default-features = false }
 utf16_iter = { version = "1.0.2", default-features = false }
 utf8_iter = { version = "1.0.2", default-features = false }
-wasmi = { version = "0.31.2", default-features = false }
 write16 = { version = "1.0.0", default-features = false }
+
+## External Deps Group 2: Heavy Dev and Datagen deps. No default features.
 zip = { version = "0.6.0", default-features = false }
+serde-aux = { version = "4.0.0", default-features = false }
 
-## Dev only
+## External Deps Group 3: Dev and Datagen deps. Include default features.
```

Comment:
I did prefer the split between dev deps and non-dev deps. For example this looks like datagen depends on `rkyv`, so someone might introduce the dep somewhere else.

Reply:
I'll land this because it fixes problems and is likely to bitrot, and if you want to propose a different organization you're welcome to do so

Reply:
Well I had this different organisation before

Reply:
You had "deps" and "dev deps" , but you didn't distinguish lib deps from datagen deps

Reply:
> For example this looks like datagen depends on rkyv, so someone might introduce the dep somewhere else.

I consider depcheck allowlist.rs to be the source of truth for what depends on what.